### DISCUSSION TITLE
Support for nextflow logs in the Logfile Navigator (a terminal-based log file viewer)

### DISCUSSION THREAD

**Discussion by tstack** [nextflow-io/nextflow#4824] [Show and tell]
Hello, I just wanted to mention that I've added basic support for nextflow logs to the Logfile Navigator (https://lnav.org).  lnav is a log file viewer for the terminal that can tail, merge, highlight, and analyze logs of a wide variety of formats.  Here is a screenshot of a nextflow log:

<img width="976" alt="Screenshot 2024-03-18 at 10 31 20 PM" src="https://github.com/nextflow-io/nextflow/assets/107532/7790dbe7-f9d3-41b5-90ef-4cdd68d843ea">

lnav uses JSON files to define a format, so you can [install lnav](https://lnav.org/downloads), download [nextflow_log.json](https://raw.githubusercontent.com/tstack/lnav/master/src/formats/nextflow_log.json), and then run the following to install the file:

```sh
lnav -i nextflow_log.json
```

Alternatively, you can download a [nightly build of lnav](https://github.com/tstack/lnav/actions/runs/8338106337) that will have the format builtin.

Viewing a file is just a matter of passing the paths/globs of the files or directories to open, like so:

```
lnav .nextflow.log
```

(lnav will automatically detect the file format, decompress files, open new files in a directory, etc...)

I'm not familiar with nextflow, so I'm not sure if some extra highlighting or other things would be useful.  If folks have suggestions, I'm happy to try and implement them.

Thanks

**Comment by pditommaso** [comment]
Nice!
